### PR TITLE
HWKAPM-813 : Mark nodes as faulty on "error" property

### PR DESCRIPTION
**Pull request by ammendonca**

### CODE REVIEW

File: ui/src/main/scripts/plugins/directives/instance-view-diagram/ts/instanceViewDiagramDirective.ts [hunk 1]
```diff
@@ -198,7 +198,7 @@ module InstanceViewDiagram {
           _.each(d.properties, (property) => {
             if (!!property.value) {
               nodeTT += ('<li class=\'tt-prop\'><strong>' + property.name + '</strong> ' + property.value + '</li>');
-              if (property.name === 'fault') {
+              if (property.name === 'fault' || property.name === 'error') {
```
Comment (objectiser) [inline ui/src/main/scripts/plugins/directives/instance-view-diagram/ts/instanceViewDiagramDirective.ts]: Sorry we should have been more precise, we need to check for property 'error' with value 'true'. Its possible that error=false may be specified. You may need to deal with different cases.

Comment (pavolloffay) [inline ui/src/main/scripts/plugins/directives/instance-view-diagram/ts/instanceViewDiagramDirective.ts]: Or be even more bulletproof by `prop.name === 'error && prop.val !== false'`? Tag value should be boolean, but what one accidentally used it with a string value?

Comment (objectiser) [inline ui/src/main/scripts/plugins/directives/instance-view-diagram/ts/instanceViewDiagramDirective.ts]: All properties are stored as text, although numerical values are also available as a number for query purposes.

Comment (pavolloffay) [inline ui/src/main/scripts/plugins/directives/instance-view-diagram/ts/instanceViewDiagramDirective.ts]: Sorry, I don't understand what do you want to say. 

What I meant is that when someone is using `error` tag can accidentally pass other values, for instance `span.setTag(Tags.ERROR.getKey(), "whatever")`. In this case we can be more predictive and assume that span is in error state and highlight it.

...when using `Tags` api it does not allow to set wrong value: `Tags.ERROR.set(span,<< boolean>>)`

Comment (objectiser) [inline ui/src/main/scripts/plugins/directives/instance-view-diagram/ts/instanceViewDiagramDirective.ts]: Yes agree - any value other than 'false' should be treated as an error.

**Comment by ammendonca** [comment]
PR updated with suggested condition.

**Comment by objectiser** [comment]
@ammendonca LGTM :+1:

**Pull request merged**